src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -181,4 +180,29 @@ void CASEServer::OnSessionEstablished(const SessionHandle & session)
                     ChipLogValueScopedNodeId(session->GetPeer()));
     PrepareForSessionEstablishment(session->GetPeer());
 }
+
+void CASEServer::SendBusyStatusReport(Messaging::ExchangeContext * ec)
+{
+    using namespace Protocols::SecureChannel;
+
+    ChipLogProgress(SecureChannel, "Already in the middle of CASE handshake, sending busy status report");
+
+    // A successful CASE handshake can take several seconds and some may time out (30 seconds or more).
+    // TODO: Come up with better estimate: https://github.com/project-chip/connectedhomeip/issues/28288
+    // For now, setting minimum wait time to 5 seconds.
+    uint16_t minimumWaitTime                            = 5 * 1000;                // milliseconds
+    constexpr uint8_t kBusyStatusReportProtocolDataSize = sizeof(minimumWaitTime); // 16-bits
+
+    auto handle = System::PacketBufferHandle::New(kBusyStatusReportProtocolDataSize, 0);
+    VerifyOrReturn(!handle.IsNull(), ChipLogError(SecureChannel, "Failed to allocate protocol data for busy status report"));
+
+    Encoding::LittleEndian::PacketBufferWriter protocolDataBufferWriter(std::move(handle));
+    protocolDataBufferWriter.Put16(minimumWaitTime);
+    handle = protocolDataBufferWriter.Finalize();
+    VerifyOrReturn(!handle.IsNull(), ChipLogError(SecureChannel, "Failed to finalize protocol data for busy status report"));
+
+    StatusReport statusReport(GeneralStatusCode::kBusy, Id, kProtocolCodeBusy, std::move(handle));
+    ec->SendStatusReport(statusReport);
```

Comment:
I would prefer to not ignore CHIP_ERRORS and instead do logging here. Then you do not need extra logging code in SendStatusReport (since you are already returning a CHIP_ERROR)

Reply:
Newly added function does return the `CHIP_ERROR`.